app/src/main/java/net/vonforst/evmap/api/nobil/NobilModel.kt

```diff
@@ -0,0 +1,234 @@
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+import kotlinx.parcelize.Parcelize
+import net.vonforst.evmap.max
+import net.vonforst.evmap.model.Address
+import net.vonforst.evmap.model.ChargeLocation
+import net.vonforst.evmap.model.Chargepoint
+import net.vonforst.evmap.model.ChargerPhoto
+import net.vonforst.evmap.model.Coordinate
+import net.vonforst.evmap.model.Cost
+import net.vonforst.evmap.model.OpeningHours
+import net.vonforst.evmap.model.ReferenceData
+import java.time.Instant
+import java.time.LocalDateTime
+
+@JsonClass(generateAdapter = true)
+data class NobilRectangleSearchRequest(
+    val apikey: String,
+    val northeast: String,
+    val southwest: String,
+    val limit: Int,
+    val action: String = "search",
+    val type: String = "rectangle",
+    val format: String = "json",
+    val apiversion: String = "3",
+    // val existingids: String
+)
+
+@JsonClass(generateAdapter = true)
+data class NobilRadiusSearchRequest(
+    val apikey: String,
+    val lat: Double,
+    val long: Double,
+    val distance: Double, // meters
+    val limit: Int,
+    val action: String = "search",
+    val type: String = "near",
+    val format: String = "json",
+    val apiversion: String = "3",
+    // val existingids: String,
+)
+
+@JsonClass(generateAdapter = true)
+data class NobilDetailSearchRequest(
+    val apikey: String,
+    val id: String,
+    val action: String = "search",
+    val type: String = "id",
+    val format: String = "json",
+    val apiversion: String = "3",
+)
+
+@JsonClass(generateAdapter = true)
+data class NobilResponseData(
+    @Json(name = "error") val error: String?,
+    @Json(name = "Provider") val provider: String?,
+    @Json(name = "Rights") val rights: String?,
+    @Json(name = "apiver") val apiver: String?,
+    @Json(name = "chargerstations") val chargerStations: List<NobilChargerStation>?
+)
+
+@JsonClass(generateAdapter = true)
+data class NobilChargerStation(
+    @Json(name = "csmd") val chargerStationData: NobilChargerStationData,
+    @Json(name = "attr") val chargerStationAttributes: NobilChargerStationAttributes
+) {
+    fun convert() = ChargeLocation(
+        chargerStationData.id,
+        "nobil",
+        HtmlCompat.fromHtml(chargerStationData.name, HtmlCompat.FROM_HTML_MODE_COMPACT).toString(),
+        chargerStationData.position,
+        Address(chargerStationData.city,
+                when (chargerStationData.landCode) {
+                    "DAN" -> "Denmark"
+                    "FIN" -> "Finland"
+                    "ISL" -> "Iceland"
+                    "NOR" -> "Norway"
+                    "SWE" -> "Sweden"
+                    else -> ""
+                },
+                chargerStationData.zipCode,
+                listOfNotNull(chargerStationData.street, chargerStationData.houseNumber).joinToString(" ")),
+        chargerStationAttributes.conn.map { createChargepointFromNobilConnection(it.value) },
+        null,
```

Comment:
I extracted the `network` information available in nobil and there are ~1500 different "networks". The field is a text field and users have entered information in many different variations (different capitalization, typos, abbreviations, alternative names, "owned by" information, ...).

Even if I only parse operator from OCPI data I get ~60 "networks" and also here there are name variations for the same actual network and some names don't even correspond to a network.

Maybe we could create a network list manually and try to match charge locations to the list items. Not a great solution since it requires manual work to update the network list and it's hard to know which networks to include in the list, especially in "foreign" countries.

Maybe we have to live without a network filter, for now

Reply:
Ah 😟 Wow, strange that that happens even with OCPI data...
Yeah, then we probably have to live without a network filter for now, and hope that Nobil can improve the data quality in the future.